### PR TITLE
cdc: add version checking for resolved ts

### PR DESCRIPTION
**Pull request by 5kbpers**
### What problem does this PR solve?

Problem Summary:
During the rolling update, the new mechanism of advancing resolved ts may do not work.

### What is changed and how it works?

Add version checking for resolved ts.

What's Changed:

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 
* cdc: add version checking for resolved ts

### CODE REVIEW

**Comment by 5kbpers** [comment]
@liuzix PTAL

File: components/cdc/src/endpoint.rs [hunk 1]
```diff
@@ -46,6 +46,8 @@ use crate::metrics::*;
 use crate::service::{CdcEvent, Conn, ConnID, FeatureGate};
 use crate::{CdcObserver, Error, Result};
 
+const FEATURE_RESOLVED_TS_STORE: Feature = Feature::require(5, 0, 0);
```
Comment (overvenus) [inline components/cdc/src/endpoint.rs:49]: CDC has its own feature gate structure, could you unify them ?

**Review by overvenus — COMMENTED** [review]
Rest LGTM

File: components/cdc/src/endpoint.rs (hunk 1 above)
Comment (5kbpers) [reply components/cdc/src/endpoint.rs:49]: They have different purposes, this is for checking the tikv cluster version.

**Review by overvenus — COMMENTED** [review]
/lgtm

**Pull request merged**